docs/site/LB3-vs-LB4-request-response-cycle.md

```diff
@@ -0,0 +1,448 @@
+and [Interceptors](https://loopback.io/doc/en/lb4/Interceptors.html) makes it a
+very powerful extension mechanism.
+
+In LoopBack 4
+[middleware.json](https://loopback.io/doc/en/lb3/middleware.json.html)
+is not required anymore because of architectural changes.
+
+#### Models
+
+In LoopBack 3, models files automatically create the corresponding REST API
+endpoints and the database query machinery (using the configured datasource).
+In LoopBack 4, model files are limited only to describing the properties of the
+data. You will have to create a corresponding
+[Repository](https://loopback.io/doc/en/lb4/Repositories.html)
+for database connectivity, and
+[controllers](https://loopback.io/doc/en/lb4/Controllers.html)
+for creating the REST API endpoint.
+
+The fact that you have to create two more artifacts along with the model to
+get a REST endpoint working might seem overly tedious at first. However, the
+separation of concerns and decoupling the functionality makes the codebase
+cleaner, easier to maintain, and much easier to customize functionality at
+various levels. This can be better appreciated as the complexity of your app
+grows.
+
+For those who are uncomfortable with the concept of having to creating a
+repository and a controller for a model, we have a component
+[@loopback/rest-crud
+](https://loopback.io/doc/en/lb4/Creating-crud-rest-apis.html)
+; with a little bit of configuration, a model file is all you will need to
+create the REST endpoints. Once your requirements outgrow what
+`@loopback/rest-crud` provides, you can implement your REST endpoints the
+idiomatic way.
+
+#### Components
+
+Components are still supported in LoopBack 4, but the concept of component
+has completely changed.
+
+In LoopBack 3, a
+[component](https://loopback.io/doc/en/lb3/LoopBack-components.html)
+is a simple Node.js module which exports a function with the signature
+`function(app, options)`. In LoopBack 4, a
+[component](https://loopback.io/doc/en/lb4/Creating-components.html)
+is a TypeScript class which can add
+[servers](https://loopback.io/doc/en/lb4/Server.html),
+[observers](https://loopback.io/doc/en/lb4/Life-cycle.html),
+[providers](https://loopback.io/doc/en/lb4/Creating-components.html#providers),
+and controllers to the app using dependency injection.
+
+LoopBack 3 components adding routes can be migrated to LoopBack 4 by moving the
+functionality to the controller of a LoopBack 4 component.
+
+Here is an example of migrating a LoopBack 3 routing component to a LoopBack 4
+component's controller.
+
+{% include code-caption.html content="server/hi-component.js" %}
+
+```js
+module.exports = (app, options) => {
+  app.get('/hi', (req, res) => {
+    res.send('Hi!');
+  });
+};
+```
+
+{% include code-caption.html content="src/components/hi.component.ts" %}
+
+```ts
+import {get} from '@loopback/rest';
+import {Component} from '@loopback/core';
+
+export class HiController {
+  @get('/hi')
+  hello(): string {
```

Comment:
It is a migration of the LB3 route on the previous section, which sends the string "Hi":

```
  app.get('/hi', (req, res) => {
    res.send('Hi!');
  });
```

Reply:
Updated to:
```
export class HiController {
  @get('/hi', {
    responses: {
      '200': {
        content: {'application/json': {schema: {type: 'string'}}},
      },
    },
  })
  hello(): string {
    return 'Hi!';
  }
}
```

Reply:
IIUC, when trying the endpoint `/hi` in explorer, the response should be a message saying 'Hi!', which means we need to describe the response spec like:

```ts
@get('/hi', {
  responses: {
      '200': {
        description: 'Todo model instance',
        // the content object may not be accurate
        content: {'application/json': {schema: {type: 'string'}}},
      },
    },
})
great() {
  return 'hi!';
}
```

otherwise it will throw error...Or did I misunderstand anything?

Reply:
Added it as you can see.